tmt/utils/jira.py

```diff
@@ -0,0 +1,132 @@
+import urllib.parse
+from collections.abc import Sequence
+from typing import TYPE_CHECKING, Any, Optional, Union, cast
+
+import fmf.utils
+
+import tmt.base
+import tmt.log
+import tmt.utils
+from tmt._compat.pathlib import Path
+from tmt.plugins import ModuleImporter
+
+if TYPE_CHECKING:
+    import jira
+
+
+import_jira: ModuleImporter['jira'] = ModuleImporter(  # type: ignore[valid-type]
+    'jira',
+    tmt.utils.ReportError,
+    "Install 'tmt+link-jira' to use the Jira linking.")
+
+
+def jira_link(
+        nodes: Sequence[Union['tmt.base.Test', 'tmt.base.Plan', 'tmt.base.Story']],
+        links: 'tmt.base.Links',
+        logger: tmt.log.Logger,
+        separate: bool = False) -> None:
+    """ Link the object to Jira issue and create the URL to tmt web service """
+    jira_module = import_jira(logger)
+
+    def create_url_params(tmt_object: tmt.base.Core) -> dict[str, Any]:
+        tmt_type = tmt_object.__class__.__name__.lower()
+        fmf_id = tmt_object.fmf_id
+
+        url_params: dict[str, Any] = {
+            'format': 'html',
+            f'{tmt_type}-url': fmf_id.url,
+            f'{tmt_type}-name': fmf_id.name
+            }
+
+        if fmf_id.path and isinstance(fmf_id.git_root, Path):
+            fmf_path = fmf_id.path.relative_to(fmf_id.git_root).resolve().as_posix()
+            fmf_path = fmf_path.removeprefix(fmf_id.git_root.as_posix())
+            url_params[f'{tmt_type}-path'] = fmf_path.lstrip('/')
```

Comment:
What exactly are you trying to achieve here? Why `fmf_id.path` cannot be directly used here?